### DISCUSSION TITLE
Questions about catalogs

### DISCUSSION THREAD

**Discussion by mmellado** [pnpm/pnpm#8409] [Q&A] [answered]
Hi

My team and I work on a variety of different monorepos and having centralized versions of dependencies has always been a struggle that requires a lot of manual attention. I was very pleased recently when reading about the catalogs feature being out, but I have a couple of questions for which I couldn't find an answer while reading the document on the pnpm website.

First of, is management of scoped dependencies. I tried to make an initial declaration of dependencies in a catalog for one of our monorepos, but every timee try to declare a scoped dependency (starting with `@`), I get an error from PNPM saying my `pnpm-worspace.yaml` file has an indentation issue. This problem is resolved by removing the `@` sign from the catalog declaration, but I'm unsure if this would still work since the name of the package is not listed under the same name in my package and my catalog.

Now, as of now, the only step we've taken towards implementing catalogs (and probably the reason I haven't found the answer to the question above), has been declaring them, meaning that all the packages in the monorepo still point to their individual package version. The reason for doing this, is that most of the packages in this monorepo are libraries that get published individually to a private Github Packages registry. I'm curious if catalogs will properly manage the version replacement of my packages when they are published. The publishing of my packages happens as part of a pipeline process in github actions using the semantic release library, and using the pnpm plugin, meaning that the command used to run the publish is `pnpm publish`. Will the command take care of publishing the libraries with the correct version, or do I have to implement a script to set the version in the libraries' package.json from `catalog:` to the actual version?

**Comment by zkochan** [comment, marked as answer]
>  I get an error from PNPM saying my pnpm-worspace.yaml file has an indentation issue. This problem is resolved by removing the @ sign from the catalog declaration, but I'm unsure if this would still work since the name of the package is not listed under the same name in my package and my catalog.

You need to use quotes. E.g.:

```yaml
catalog:
  "@myorg/react": ^18.3.1
```

> I'm curious if catalogs will properly manage the version replacement of my packages when they are published.

It will: https://pnpm.io/catalogs#publishing

**Reply by douglasg14b** [reply]
Also ran into this, took me ages to figure it out. A more helpful error may be needed?
